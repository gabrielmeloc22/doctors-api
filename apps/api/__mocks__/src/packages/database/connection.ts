import { drizzle } from "drizzle-orm/node-postgres";

export const db = drizzle.mock({ casing: "camelCase" });
