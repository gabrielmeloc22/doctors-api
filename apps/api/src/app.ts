import bodyParser from "body-parser"
import cors from "cors"
import express, { type Express } from "express"
import morgan from "morgan"

export const createApp = (): Express => {
    const app = express()

    app.disable("x-powered-by")
        .use(morgan("dev"))
        .use(bodyParser.urlencoded({ extended: true }))
        .use(bodyParser.json())
        .use(cors())

    return app
}
