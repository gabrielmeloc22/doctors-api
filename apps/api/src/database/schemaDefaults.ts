import { randomUUID } from "crypto";
import * as t from "drizzle-orm/pg-core";

export const idField = {
    id: t.uuid().primaryKey().$defaultFn(randomUUID),
};

export const timestampField = {
    createdAt: t.timestamp().notNull().defaultNow(),
    updatedAt: t
        .timestamp()
        .notNull()
        .defaultNow()
        .$defaultFn(() => new Date()),
};
