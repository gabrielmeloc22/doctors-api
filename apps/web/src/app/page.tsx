"use client"

import { ReactNode } from "react"

export default function Web(): ReactNode {
    return <h1>batata</h1>
}
