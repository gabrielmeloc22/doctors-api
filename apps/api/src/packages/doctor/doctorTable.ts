import * as t from "drizzle-orm/pg-core";
import { idField, timestampField } from "../../database/schemaDefaults";

export const doctorTable = t.pgTable("doctor", {
    username: t.varchar({ length: 256 }).notNull().unique(),
    email: t.varchar({ length: 256 }).notNull().unique(),
    firstName: t.varchar({ length: 256 }).notNull(),
    lastName: t.varchar({ length: 256 }).notNull(),
    ...timestampField,
    ...idField,
});

export type IDoctor = typeof doctorTable.$inferSelect;
