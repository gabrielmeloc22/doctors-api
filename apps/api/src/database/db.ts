import { drizzle } from "drizzle-orm/node-postgres";
import * as booking from "../packages/booking/bookingTable";
import * as doctor from "../packages/doctor/doctorTable";
import * as slot from "../packages/slot/slotTable";

export const schema = {
    ...slot,
    ...doctor,
    ...booking,
};

export const db = drizzle({
    connection: process.env.DATABASE_URL,
    casing: "camelCase",
    schema,
});
