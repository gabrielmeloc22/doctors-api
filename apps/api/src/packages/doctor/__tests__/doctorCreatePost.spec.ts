import request from "supertest";
import { expect, it } from "vitest";
import { createApp } from "../../../app";
import { db } from "../../../database/db";
import {
    defaultFrozenKeys,
    sanitizeTestObject,
} from "../../../testUtils/sanitizeTestObject";
import { doctorTable } from "../doctorTable";

it("should post, create a new doctor and return 201", async () => {
    const app = createApp();

    const payload = {
        first_name: "John",
        last_name: "Doe",
        username: "doc007",
        email: "doc@mail.com",
    };

    const response = await request(app.listen())
        .post("/doctors")
        .send(payload)
        .expect(201);

    const doctors = await db.select().from(doctorTable);
    const doctor = doctors[0];

    const doctorResult = {
        id: doctor?.id,
        username: doctor?.username,
        first_name: doctor?.firstName,
        last_name: doctor?.lastName,
        email: doctor?.email,
    };

    const body = response.body as Record<string, unknown>;

    expect(doctors.length).toBe(1);

    expect(doctorResult).toMatchObject(payload);

    expect(body).toMatchObject({
        doctor: doctorResult,
        success: true,
    });

    expect(
        sanitizeTestObject({
            obj: body,
            frozenKeys: defaultFrozenKeys,
        })
    ).toMatchSnapshot();
});
