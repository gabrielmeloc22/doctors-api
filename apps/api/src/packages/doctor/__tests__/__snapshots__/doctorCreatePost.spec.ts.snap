// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should post, create a new doctor and return 201 1`] = `
{
  "doctor": {
    "email": "doc@mail.com",
    "first_name": "John",
    "id": "1b3cffbc-0f1a-4d43-ab5f-71ca3db21b91",
    "last_name": "Doe",
    "username": "doc007",
  },
  "success": true,
}
`;
